File: CSharp/leetcodesln/XORQueriesOfASubarray.cs
namespace leetcodesln
{
    public class XORQueriesOfASubarray
    {
        public int[] XorQueries(int[] arr, int[][] queries)
        {
            var prefixXOR = new int[arr.Length + 1];

            for (int i = 1; i < prefixXOR.Length; i++)
            {
                prefixXOR[i] = prefixXOR[i - 1] ^ arr[i - 1];
            }

            var ans = new int[queries.Length];

            for (int i = 0; i < queries.Length; i++)
            {
                var l = queries[i][0];
                var r = queries[i][1];

                ans[i] = prefixXOR[r + 1] ^ prefixXOR[l];
            }
            return ans;

        }
    }
}

File: Go/questions/XORQueriesOfASubarray.go
package questions

func XorQueries(arr []int, queries [][]int) []int {
	ans := make([]int, len(queries))
	preTable := make([]int, len(arr)+1)

	for i := 0; i < len(arr); i++ {
		preTable[i+1] = preTable[i] ^ arr[i]
	}

	for i := 0; i < len(queries); i++ {
		l := queries[i][0]
		r := queries[i][1]
		ans[i] = preTable[r+1] ^ preTable[l]
	}
	return ans
}
